### PR TITLE
[MRG] adding support for apple health

### PR DESCRIPTION
**Pull request by gedankenstuecke**
This PR:
- adds the frontend features to display our new support for Apple Watch/Apple Health
- adds the parsers needed on the backend
- adds those parsers to the `RetrospectiveEvent` and `SymptomReport` objects as is done for all other wearables. 

I've tested it with my data and it worked for all cases. I think it should be ready to be deployed!

### CODE REVIEW

File: retrospective/tasks.py [hunk 1]
```diff
@@ -153,15 +175,18 @@ def analyze_event(event_id):
...
+                    fitbit_data, fitbit_intraday_data, event.date
+                )
+                new_analysis = RetrospectiveEventAnalysis(
+                    event=event,
+                    graph_data=json.dumps(fb_intraday_data),
+                    graph_type="fitbit_intraday",
+                )
+                new_analysis.save()
+            except:
+                pass
```
Comment (carolinux) [inline retrospective/tasks.py:189]: in which case would an exception be thrown here? maybe add a comment

File: quantified_flu/helpers.py [hunk 2]
```diff
@@ -9,16 +9,25 @@
...
             missing_sources.pop("oura", None)
         if i["basename"] == "fitbit-data.json" and i["source"] == "direct-sharing-102":
             missing_sources.pop("fitbit", None)
-        if "googlefit" in i["basename"]: # GoogleFit is stored in monthly files
+        if "googlefit" in i["basename"]:  # GoogleFit is stored in monthly files
             missing_sources.pop("googlefit", None)
+        if i["source"] == "direct-sharing-453" and i["basename"].startswith(
+            "heartrate_samples"
+        ):
+            missing_sources.pop("apple_health", None)
```
Comment (carolinux) [inline quantified_flu/helpers.py:30]: Alright so this PR handles the case of the data having been imported by OH already, and doesn't offer an option to import them through QF. Is this planned for future?

File: retrospective/tasks.py (hunk 1 above)
Comment (gedankenstuecke) [reply retrospective/tasks.py:189]: Good point, it's a rather general catch-all as the merging of Fitbit & Fitbit Intraday is really fickle and easily crashes (e.g. I stopped using my Fitbit some time ago but still have my data in my account. Fitbit still updates the data even and reports `0` for each day, this leads the whole pipeline to crash 🙈 ).

File: quantified_flu/helpers.py (hunk 2 above)
Comment (gedankenstuecke) [reply quantified_flu/helpers.py:30]: Nope, the `Apple Health` data is a bit special, as you can't export it through an API or anything, but just through having the corresponding iPhone app installed that exports data. Which makes integrating it much harder.

To integrate this in QF we'd need to generate a whole new API to which to export the data (not to mention the fact that it took 2 months to get the app into the appstore :D). 

So using the app we got now seems the way to go (plus it doesn't make much difference, as people need to use an external tool in any case).

Comment (carolinux) [reply quantified_flu/helpers.py:30]: I see :D

File: quantified_flu/templates/quantified_flu/home.html [hunk 3]
```diff
@@ -203,6 +214,26 @@ <h3>Manage Oura</h3>
       <a href="https://oura.openhumans.org">oura.openhumans.org</a>
       </p>
       {% endif %}
+      </div>
+      {% endif %}
```
Comment (madprime) [inline quantified_flu/templates/quantified_flu/home.html:218]: I opened up the code wondering if this was an extra endif (it's not) & have a minor code formatting PR to share. just readability, as I was trying to check this over :)

File: retrospective/tasks.py (hunk 1 above)
Comment (madprime) [reply retrospective/tasks.py:191]: not ideal but better than a server error ruining other processing/response? wouldn't hold this up based on improving this further

File: quantified_flu/helpers.py (hunk 2 above)
Comment (gedankenstuecke) [reply quantified_flu/helpers.py:30]: Thanks @madprime!

**Pull request merged**